rust/lance-encoding/src/encodings/physical/packed_struct.rs

```diff
@@ -0,0 +1,405 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use std::sync::Arc;
+
+use arrow_array::{Array, ArrayRef, StructArray};
+use futures::{future::BoxFuture, FutureExt};
+use lance_arrow::DataTypeExt;
+
+use crate::{
+    decoder::{PageScheduler, PrimitivePageDecoder},
+    encoder::{ArrayEncoder, EncodedArray, EncodedArrayBuffer},
+    format::pb::{self},
+    EncodingsIo,
+};
+
+use arrow_buffer::buffer::Buffer;
+use arrow_schema::DataType;
+use bytes::Bytes;
+use bytes::BytesMut;
+use lance_core::Result;
+
+#[derive(Debug)]
+pub struct PackedStructPageScheduler {
+    _inner_schedulers: Vec<Box<dyn PageScheduler>>,
```

Comment:
Yes, because we can just use `submit_request()` to decode the entire chunk of bytes and don't need the inner decoders
If we later use more complicated encodings like bitpacking it can be potentially useful, so I didn't remove it. I have added a comment explaining this